libevmcore/EVMSchedule.h

```diff
@@ -108,6 +109,7 @@ static const EVMSchedule EIP158Schedule = []
 static const EVMSchedule MetropolisSchedule = []
 {
 	EVMSchedule schedule = EIP158Schedule;
+	schedule.blockhashGas = 800;
```

Comment:
Yes, this seem to be strange. Shouldn't we just charge for the contract execution?